tests/sentry/api/endpoints/test_project_rules.py

```diff
@@ -2,35 +2,54 @@
 from unittest.mock import patch
 
 import responses
-from django.urls import reverse
 
 from sentry.models import Environment, Integration, Rule, RuleActivity, RuleActivityType
 from sentry.testutils import APITestCase
 from sentry.utils import json
 
 
-class ProjectRuleListTest(APITestCase):
-    def test_simple(self):
-        self.login_as(user=self.user)
-
-        team = self.create_team()
-        project1 = self.create_project(teams=[team], name="foo")
-        self.create_project(teams=[team], name="bar")
+class ProjectRuleBaseTestCase(APITestCase):
+    endpoint = "sentry-api-0-project-rules"
 
-        url = reverse(
-            "sentry-api-0-project-rules",
-            kwargs={"organization_slug": project1.organization.slug, "project_slug": project1.slug},
+    def setUp(self):
+        self.user = self.create_user()
+        self.organization = self.create_organization(owner=self.user)
+        self.project = self.create_project(organization=self.organization)
+        self.rule = self.create_project_rule(project=self.project)
+        self.slack_integration = Integration.objects.create(
+            provider="slack",
+            name="Awesome Team",
+            external_id="TXXXXXXX1",
+            metadata={
+                "access_token": "xoxp-xxxxxxxxx-xxxxxxxxxx-xxxxxxxxxxxx",
+                "installation_type": "born_as_bot",
+            },
+        )
+        self.slack_integration.add_organization(self.organization, self.user)
+        self.sentry_app = self.create_sentry_app(
+            name="Pied Piper",
+            organization=self.organization,
+            schema={"elements": [self.create_alert_rule_action_schema()]},
+        )
+        self.sentry_app_installation = self.create_sentry_app_installation(
+            slug=self.sentry_app.slug, organization=self.organization
         )
-        response = self.client.get(url, format="json")
+        self.sentry_app_settings_payload = [
+            {"name": "title", "value": "Team Rocket"},
+            {"name": "summary", "value": "We're blasting off again."},
+        ]
+        self.login_as(user=self.user)
 
-        assert response.status_code == 200, response.content
 
-        rule_count = Rule.objects.filter(project=project1).count()
-        assert len(response.data) == rule_count
+class ProjectRuleListTest(ProjectRuleBaseTestCase):
+    def test_simple(self):
+        response = self.get_success_response(
+            self.organization.slug, self.project.slug, status_code=200
+        )
+        assert len(response.data) == Rule.objects.filter(project=self.project).count()
 
 
-class CreateProjectRuleTest(APITestCase):
-    endpoint = "sentry-api-0-project-rules"
+class CreateProjectRuleTest(ProjectRuleBaseTestCase):
     method = "post"
 
     def run_test(
```

Comment:
It's weird that this works without a default value for `actions`. I'd to 
```py
def run_test(
    self,
    actions: Sequence[Mapping[str, Any]] | None = None,
    conditions: Sequence[Mapping[str, Any]] | None = None,
    filters: Sequence[Mapping[str, Any]] | None = None,
    name: str | None = "hello world",
    action_match: str | None = "any",
    filter_match: str | None = "any",
    frequency: int | None = 30,
    expected_conditions: Sequence[Mapping[str, Any]] | None = None,
    **kwargs: Any,
) -> None:
```
to be safe.